nemo/collections/multimodal/speech_llm/models/modular_models.py

```diff
@@ -1726,6 +1759,67 @@ def find_frozen_submodules(model):
             self.perception = self.trainer.strategy._setup_model(self.perception)
             self.perception = self.perception.cuda(torch.cuda.current_device())
 
+    def oomptimizer_schema(self, schema: str = "audio") -> dict:
+        """
+        Return a typing schema for optimal batch size calibration for various
```

Comment:
can u elaborate a little bit on how to write up this function if we want to add other schema in future

Reply:
Sure. This is a description of what keys and values to expects in the mini-batch dict returned from dataloader. We expect different items for audio and text so you can select the one you want. We are using this inside `oomptimizer.py` to generate random data for checking OOM. I can write a separate tutorial for this.